### PR TITLE
feat(Stream): race

### PR DESCRIPTION
**Pull request by dilame**

### CODE REVIEW

File: packages/effect/src/Stream.ts [hunk 1]
```diff
@@ -3472,6 +3472,42 @@ export const provideSomeLayer: {
...
+ * @since 3.6.0
+ * @category racing
+ */
+export const race: {
+  <A2, E2, R2>(
+    stream2: Stream<A2, E2, R2>
+  ): <A1, E1, R1>(stream1: Stream<A1, E1, R1>) => Stream<A1 | A2, E1 | E2, R1 | R2>
+  <A1, E1, R1, A2, E2, R2>(
+    stream1: Stream<A1, E1, R1>,
+    stream2: Stream<A2, E2, R2>
```
Comment (tim-smart) [inline packages/effect/src/Stream.ts:3507]: To align with the other apis:

```suggestion
    that: Stream<A2, E2, R2>
  ): <A1, E1, R1>(self: Stream<A1, E1, R1>) => Stream<A1 | A2, E1 | E2, R1 | R2>
  <A1, E1, R1, A2, E2, R2>(
    self: Stream<A1, E1, R1>,
    that: Stream<A2, E2, R2>
```

**Review by tim-smart — CHANGES_REQUESTED** [review]

**Comment by dilame** [comment]
@tim-smart i applied the changes from the review

File: packages/effect/src/Stream.ts [hunk 2]
```diff
@@ -3472,6 +3472,42 @@ export const provideSomeLayer: {
...
+ *
+ * Effect.runPromise(Stream.runDrain(stream))
+ * // Output each millisecond from the first stream, the rest streams are interrupted
+ * // 0
+ * // 1
+ * // 2
+ * // 3
+ * // 4
+ * // 5
+ * @since 3.6.0
```
Comment (tim-smart) [inline packages/effect/src/Stream.ts:3498]:
```suggestion
 * @since 3.7.0
```

**Review by tim-smart — CHANGES_REQUESTED** [review]

**Review by tim-smart — APPROVED** [review]

**Pull request merged**